include/highfive/H5DataType.hpp

```diff
@@ -158,6 +158,67 @@ class CompoundType : public DataType {
     void create(size_t size = 0);
 };
 
+///
+/// \brief Create a enum HDF5 datatype
+///
+/// \code{.cpp}
+/// enum class Position {
+///     FIRST = 1,
+///     SECOND = 2,
+/// };
+///
+/// EnumType<Position> create_enum_position() {
+///     return EnumType<Position>({{"FIRST", Position::FIRST},
+///                                {"SECOND", Position::SECOND}});
+/// }
+///
+/// template<>
+/// DataType create_datatype<Position>() {
+///     return create_enum_position();
+/// }
+///
+/// void write_first(H5::File& file) {
+///     auto en = create_enum_position();
+///     en.commit(file, "Position");
+///     auto dataset = file.createDataset("/foo", DataSpace(1), en);
+///     dataset.write(Position::FIRST);
+/// }
+/// \endcode
+template<typename T>
+class EnumType: public DataType {
+public:
+    ///
+    /// \brief Use for defining a member of enum type
+    struct member_def {
+        member_def(std::string t_name, T t_value)
```

Comment:
Prefer const-ref argument which gets copied. Cheaper than copy argument and then move. If you want to be faster implement also a move constructor `std::string&&` which you can move